### PR TITLE
EPMRPP-60627  Adopted Make decision modal to the step level

### PR DESCRIPTION
**Pull request by katrin-kot**

### CODE REVIEW

File: app/src/pages/inside/stepPage/modals/editDefectModals/makeDecisionModal/makeDecisionModal.jsx [hunk 1]
```diff
@@ -41,9 +41,10 @@ const MakeDecision = ({ data }) => {
   const { formatMessage } = useIntl();
   const dispatch = useDispatch();
   const activeProject = useSelector(activeProjectSelector);
-  const itemData = data.item;
+  const itemsData = data.items && data.items.length === 1 ? data.items[0] : data.items;
+  const itemData = data.item ? data.item : itemsData;
```
Comment (KatsiarynaTatarynovich) [inline app/src/pages/inside/stepPage/modals/editDefectModals/makeDecisionModal/makeDecisionModal.jsx:45]: You can simplify this expression as follows
`data.item || itemsData`

**Review by KatsiarynaTatarynovich — CHANGES_REQUESTED** [review]

File: app/src/pages/inside/stepPage/stepPage.jsx [hunk 2]
```diff
@@ -358,7 +359,7 @@ export class StepPage extends Component {
     });
     if (isDefectGroupOperation) {
```
Comment (AmsterGet) [inline app/src/pages/inside/stepPage/stepPage.jsx:360]: Now it can be simplified, show just one modal here (MAKE_DECISION_MODAL)

**Review by AmsterGet — CHANGES_REQUESTED** [review]

File: app/src/pages/inside/stepPage/modals/editDefectModals/makeDecisionModal/makeDecisionModal.jsx [hunk 3]
```diff
@@ -41,9 +41,10 @@ const MakeDecision = ({ data }) => {
   const { formatMessage } = useIntl();
   const dispatch = useDispatch();
   const activeProject = useSelector(activeProjectSelector);
-  const itemData = data.item;
+  const itemsData = data.items && data.items.length === 1 ? data.items[0] : data.items;
```
Comment (AmsterGet) [inline app/src/pages/inside/stepPage/modals/editDefectModals/makeDecisionModal/makeDecisionModal.jsx:44]: This can be simplified by providing just an array with items in different cases (pass `items: [item]` for single operation or `items: items` for bulk operation).

**Review by AmsterGet — CHANGES_REQUESTED** [review]

File: app/src/pages/inside/stepPage/modals/editDefectModals/makeDecisionModal/makeDecisionModal.jsx (hunk 3 above)
Comment (katrin-kot) [reply app/src/pages/inside/stepPage/modals/editDefectModals/makeDecisionModal/makeDecisionModal.jsx:44]: I further use the logic of checking 'itemData' if it is an array.

File: app/src/pages/inside/stepPage/stepPage.jsx [hunk 4]
```diff
@@ -455,7 +418,6 @@ export class StepPage extends Component {
             onUnselectAll={this.unselectAllItems}
             onProceedValidItems={this.proceedWithValidItems}
             onRefresh={this.props.fetchTestItemsAction}
-            debugMode={debugMode}
```
Comment (AmsterGet) [inline app/src/pages/inside/stepPage/stepPage.jsx:457]: Why did you remove this?

File: app/src/pages/inside/stepPage/stepPage.jsx [hunk 5]
```diff
@@ -46,7 +45,6 @@ import {
   editDefectsAction,
   linkIssueAction,
   postIssueAction,
-  isDefectGroupOperationAvailable,
```
Comment (AmsterGet) [inline app/src/pages/inside/stepPage/stepPage.jsx:49]: You can completely delete this utility if it is not used anywhere.

**Review by AmsterGet — CHANGES_REQUESTED** [review]

File: app/src/pages/inside/stepPage/stepPage.jsx (hunk 5 above)
Comment (katrin-kot) [reply app/src/pages/inside/stepPage/stepPage.jsx:49]: It uses in component ActionPanelWithGroupOperations.

File: app/src/pages/inside/stepPage/stepPage.jsx (hunk 4 above)
Comment (katrin-kot) [reply app/src/pages/inside/stepPage/stepPage.jsx:457]: Yes, I deleted it by mistake

File: app/src/pages/inside/historyPage/historyToolbar/actionPanelWithGroupOperations/actionPanelWithGroupOperations.jsx [hunk 6]
```diff
@@ -289,46 +288,18 @@ export class ActionPanelWithGroupOperations extends Component {
...
+      fetchFunc: this.unselectAndRefreshItems,
       debugMode,
+      eventsInfo: {
+        editDefectsEvents: HISTORY_PAGE_EVENTS.EDIT_DEFECT_MODAL_EVENTS,
+        unlinkIssueEvents: UNLINK_ISSUE_EVENTS_INFO,
+        postIssueEvents: POST_ISSUE_EVENTS_INFO,
+        linkIssueEvents: LINK_ISSUE_EVENTS_INFO,
+      },
     });
-    tracking.trackEvent(HISTORY_PAGE_EVENTS.EDIT_DEFECT_ACTION);
```
Comment (AmsterGet) [inline app/src/pages/inside/historyPage/historyToolbar/actionPanelWithGroupOperations/actionPanelWithGroupOperations.jsx:300]: Please keep this event tracking.

**Review by AmsterGet — CHANGES_REQUESTED** [review]

**Comment by AmsterGet** [comment]
Also please resolve conflicts

**Review by AmsterGet — APPROVED** [review]
